src/plone/restapi/tests/test_content_local_roles.py

```diff
@@ -38,6 +38,25 @@ def _get_ac_local_roles_block(self, obj):
                     '__ac_local_roles_block__',
                     False))
 
+    def test_sharing_search(self):
+        '''A request to @sharing should support the search parameter. '''
+        response = requests.get(
+            self.portal.folder1.absolute_url() + '/@sharing',
+            headers={'Accept': 'application/json'},
+            auth=(SITE_OWNER_NAME, SITE_OWNER_PASSWORD),
+        )
+        non_search_entries = response.json()['entries']
+
+        response = requests.get(
+            self.portal.folder1.absolute_url() + '/@sharing?search=admin',
+            headers={'Accept': 'application/json'},
+            auth=(SITE_OWNER_NAME, SITE_OWNER_PASSWORD),
+        )
+        search_entries = response.json()['entries']
+
+        # Did we find anything?
+        self.assertNotEqual(len(non_search_entries), len(search_entries))
```

Comment:
@jaroel could we add some more tests here to make it visible what the sharing search actually returns?